packages/snip20/src/query.rs

```diff
@@ -0,0 +1,315 @@
+pub struct Allowance {
+    pub spender: HumanAddr,
+    pub owner: HumanAddr,
+    pub allowance: Uint128,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub expiration: Option<u64>,
+}
+/// Balance response
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct Balance {
+    pub amount: Uint128,
+}
+/// Transaction data
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct Tx {
+    pub id: u64,
+    pub from: HumanAddr,
+    pub sender: HumanAddr,
+    pub receiver: HumanAddr,
+    pub coins: Coin,
+}
+/// TransferHistory response
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct TransferHistory {
+    pub txs: Vec<Tx>,
+}
+/// Minters response
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct Minters {
+    pub minters: Vec<HumanAddr>,
+}
+
+/// SNIP20 queries
+#[derive(Serialize)]
+#[serde(rename_all = "snake_case")]
+pub enum Snip20QueryMsg<'a> {
+    TokenInfo {},
+    ExchangeRate {},
+    Allowance {
+        owner: &'a HumanAddr,
+        spender: &'a HumanAddr,
+        key: &'a str,
+    },
+    Balance {
+        address: &'a HumanAddr,
+        key: &'a str,
+    },
+    TransferHistory {
+        address: &'a HumanAddr,
+        key: &'a str,
+        page: Option<u32>,
+        page_size: u32,
+    },
+    Minters {},
+}
+
+impl<'a> fmt::Display for Snip20QueryMsg<'a> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match *self {
+            Snip20QueryMsg::TokenInfo { .. } => write!(f, "TokenInfo"),
+            Snip20QueryMsg::ExchangeRate { .. } => write!(f, "ExchangeRate"),
+            Snip20QueryMsg::Allowance { .. } => write!(f, "Allowance"),
+            Snip20QueryMsg::Balance { .. } => write!(f, "Balance"),
+            Snip20QueryMsg::TransferHistory { .. } => write!(f, "TransferHistory"),
+            Snip20QueryMsg::Minters { .. } => write!(f, "Minters"),
+        }
+    }
+}
+
+impl<'a> Snip20QueryMsg<'a> {
+    /// Returns a StdResult<T>, where T is the "Response" type that wraps the query answer
+    ///
+    /// # Arguments
+    ///
+    /// * `deps` - reference to Extern that holds all the external contract dependencies
+    /// * `block_size` - pad message to blocks of this size
+    /// * `callback_code_hash` - string slice holding the code hash of contract being queried
+    /// * `contract_addr` - reference to address of contract being queries
+    pub fn query<S: Storage, A: Api, Q: Querier, T: DeserializeOwned>(
+        &'a self,
+        deps: &'a Extern<S, A, Q>,
+        block_size: usize,
+        callback_code_hash: &'a str,
+        contract_addr: &'a HumanAddr,
+    ) -> StdResult<T> {
+        let pad_block_size: usize;
+        // can not have block size of 0
+        if block_size == 0 {
+            pad_block_size = 1;
+        } else {
+            pad_block_size = block_size;
+        }
```

Comment:
if you add `mut` to the declaration of `block_size` in the function signature, you can just write:
```rust
if block_size == 0 {
    block_size = 1;
}
```